### ISSUE TITLE
Disable HTML description generation for OCA repositories ?

### ISSUE THREAD

**Issue by legalsylvain** [OCA/maintainer-tools#399]
regarding https://github.com/OCA/maintainer-tools/pull/397#issuecomment-442800276

Pro : @pedrobaeza 
> Maybe now that we have our apps store, we should disable the HTML generation?

Cons : @elicoidal 
> We should focus in our apps store but cannot ignore the official one where for the moment most people go.
All the more that before it is actually nice enough: we need to add images in all modules and improve a lot of README description.
> I still would keep versatility whenever possible: being in the 2 stores increases our visibility and SEO
> Today without HTML desc the odoo apps store is rating the module badly so I vote to keep it

**Comment by pedrobaeza** [comment]
My argument to disable them are:

* It bloats a lot the noise on PRs, the repository size and duplicated info, that are already duplicated by pieces in `readme` folder, so better to avoid a third duplication.
* We are not having "SEO" advantages right now nor expected in the future on Odoo apps store although we generate that HTML, as we can see in https://github.com/OCA/maintainer-tools/pull/397#issuecomment-442800276

**Comment by legalsylvain** [comment]
Naive question : is it possible that having html files will generate more possible conflict, when porting / cherry pickinging patches / etc .. ?

otherwise, i abstain on that point. I agree with @pedrobaeza in a technical point of view. (huge html files sucks and are useless). In the other hand, I understand "political" position of @elicoidal. The mission of the OCA *"is to promote the widespread use of Odoo and to support the collaborative development of Odoo features."* (https://odoo-community.org/)

**Comment by elicoidal** [comment]
I reckon HTML adds much more powerful content than README but we currently create it one-to-one from README so no content advantage of the automatic generation.
Not sure there are additional drawbacks

**Comment by pedrobaeza** [comment]
RST can be the same powerful as HTML, due to HTML is sanitized on apps store, so no possible JS scripting or powerful features, so HTML is going to be always 1:1 replaceable by RST.

**Comment by elicoidal** [comment]
Check this: https://www.odoo.com/apps/modules/12.0/base_module_record/
This is not possible to easily achieve in rst.
Anyway, as I said, as we actually automatically generate the html from rst, there is no interest to html as it will always be the same as the rst.
Who wants a nice looking html will do it manually IMO

So in summary, we might not need auto-generated html.
cc @OCA/board

**Comment by sbidoul** [comment]
I'm neutral on this topic.

Note to self: the html generation validates the rst as a side effect, so if we drop it, we need to keep the validation part.

**Comment by pedrobaeza** [comment]
@elicoidal well, that kind of decoration indeed not, but it's something that we are not going to do any way.

@sbidoul we should move the validation any way to Travis CI in PR stage, as now we can have the problem that the CI fails after merging.

**Comment by legalsylvain** [comment]
>  Who wants a nice looking html will do it manually IMO

Well, if we accept custom html in the OCA. (for the time being, I didn't saw any in OCA repositories ..) We *should* disable auto-generation of html index, otherwise, it will erase the custom index.
I personnaly not in favor to promote custom html files in the OCA. It is harder to review, (and make another thing *not very interesting* to review)

**Comment by elicoidal** [comment]
> Well, if we accept custom html in the OCA.

Currently we accept custom README and see no reason why not accepting custom html too:
* It can bring real nice description to the shop.
* Another point is that we could work on the css for the automatic generation to make it really nice looking and sellable.
* same as rst: if custom index.html exists (or key), we skip the auto-generation.
* Nevertheless you are right: this brings extra burden to the reviewers. I would suggest to leave it our of the current review (or only light review) and trust the contributor to stick to simple rules

**Comment by legalsylvain** [comment]
> Currently we accept custom README and see no reason why not accepting custom html too:

You're right, I didn't see this line. https://github.com/OCA/maintainer-tools/blob/master/tools/gen_addon_readme.py#L194
Custom HTML will not be erased. 

> and sellable.

Could you detail your point of view ? Are you suggesting that the OCA business model could be based on the sell of apps ?
